### DISCUSSION TITLE
Github action always freeze at signing step??

### DISCUSSION THREAD

**Discussion by jim-jiang-github** [AvaloniaUI/Avalonia#11823] [Q&A]
My repo: https://github.com/jim-jiang-github/LogVisualizer
Branch: macos_ci_1
Script at: ./scripts/macos/signing.sh
Job: https://github.com/jim-jiang-github/LogVisualizer/actions/runs/5303433040/jobs/9599052326

Every thing works well on my device:
<img width="570" alt="image" src="https://github.com/AvaloniaUI/Avalonia/assets/29189410/70c6d8f8-64f8-4772-a727-49794e6dfac0">

But always freeze on github action job, Can someone met same issue?
<img width="1211" alt="image" src="https://github.com/AvaloniaUI/Avalonia/assets/29189410/162d1f9f-56c4-4d57-b77c-e23401ffdde0">

Can not go on signing step.
<img width="1356" alt="image" src="https://github.com/AvaloniaUI/Avalonia/assets/29189410/5a1e954f-9408-487d-8cf3-611152cfb9ca">

**Comment by rabbitism** [comment]
Your ci file path looks one level deeper than your local result.

**Reply by jim-jiang-github** [reply]
If there is an additional layer in the CI file path compared to the local result, it should typically result in an error rather than a freezing of the process, right?

**Reply by timunie** [reply]
i don't see how this is Avalonia related tbh. Am I missing something?

**Reply by jim-jiang-github** [reply]
I defined some scripts in the .yml file.
https://github.com/jim-jiang-github/LogVisualizer/blob/macos_ci_1/.github/workflows/ci.yml

**Reply by kekekeks** [reply]
signtool is likely to be asking for keyring password using the macos UI. You might need to click on the dialog to make it work, I believe.

**Reply by jim-jiang-github** [reply]
Any solution?

**Reply by kekekeks** [reply]
Enter the keyring password on the CI agent

**Reply by kekekeks** [reply]
It might also have problems with accessing apple servers or signature timestamp servers
